shell/components/AppModal.vue

```diff
@@ -150,19 +170,12 @@ export default defineComponent({
         };
       }
 
-      if (!props.triggerFocusTrapWatcherBased) {
-        useBasicSetupFocusTrap('#modal-container-element', opts);
+      // needed as a workaround/fix for PromptModal unit tests
+      if (props.fixUnitTestTrapOptions) {
+        opts.tabbableOptions = { displayCheck: 'none' };
       }
-    }
-  },
-  created() {
-    // This usecase is to cover the PromptModal scenario where it's renders a generic component inside.
-    // Due to the architecture of the PromptModal it needs to be handled with a watcher based focus trap
-    // but with a dedicated unmount hook
-    if (this.triggerFocusTrapWatcherBased) {
-      const opts:any = DEFAULT_FOCUS_TRAP_OPTS;
 
-      useWatcherBasedSetupFocusTrapWithDestroyIncluded(() => this.focusTrapWatcherBasedVariable, '#modal-container-element', opts, true);
+      useWatcherBasedSetupFocusTrapWithDestroyIncluded(() => props.focusTrapWatcherBasedVariable !== undefined ? props.focusTrapWatcherBasedVariable : props.fakeWatchVar, '#modal-container-element', opts, true);
```

Comment:
Can this be simplified to the following?

```suggestion
      useWatcherBasedSetupFocusTrapWithDestroyIncluded(        props.focusTrapWatcherBasedVariable ?? props.fakeWatchVar,, '#modal-container-element', opts, true);
```

In which scenarios would the `focusTrapWatcherBasedVariable` be omitted but the `fakeWatchVar` is provided?

Reply:
Ah, yes... The nullish coalescing operator is a much better fit here, you're right. It does need the function though. 

That's needed because of why the `watch` works in the composition API https://vuejs.org/guide/essentials/watchers.html check the getter function usage